### PR TITLE
[FIX] web_editor: allow mass mailing users to access the editor

### PR DESCRIPTION
**Pull request by qsm-odoo**
Mass mailing users with low access rights were not able to access the
editor anymore if the website application was not installed (or any
other app giving ir.ui.view read access rights to the related users).

This commit fixes the problem but should be handled another way in a
future master refactoring. The problem was that the views are now
private by default: they can be read thanks to read_template provided
they declare a group that the user has and that the user has read access
rights.

For the case 'read_template web_editor.colorpicker', it works for
website editor users as the view has the base.group_user group
*and they have access rights thanks to publisher/designer groups*.
For mass mailing users, no such group exists though so they simply
do not have the rights to read that template anymore. Seems safer
to force it for this template only while waiting for a better
access rights refactoring.

Note: using 'render_public_asset' which allows to bypass rights if
the user has the group the view requires was also a solution.
However, that would turn the 'read' into a 'render', which is
a less stable change.

task-2412544

### CODE REVIEW

**Comment by qsm-odoo** [comment]
@mart-e What I did here is obviously a solution for stable but I wonder why `render_public_asset` bypasses access rights if the view declares a group and that the user matches it... but not `read_template` (which still only allows to read the views which declare a group that user can match... but still require read access rights to be read).

Side note: not convinced about the word 'asset' in that method name which is really confusing with our assets system... :grimacing: and between `public_render_template`, `read_template`, `render_public_asset` ... which all act differently regarding authorization and access rights, this calls for a refactoring in master ^^

**Comment by mart-e** [comment]
> I wonder why `render_public_asset` bypasses access rights if the view declares a group and that the user matches it... but not `read_template` (which still only allows to read the views which declare a group that user can match... but still require read access rights to be read).

TL;DR : can you create a controller just for this view?

The rational was that reading and rendering arbitrary view content should be avoided (as it may content some specific business logic or be used as `safe_eval` vector). To achieve that, we remove global access and kept a few helpers to cover the most common scenarios.

A few views needed to be rendered based on the external id by low priviledge users, hence the whitelisting. This is not ideal but was the easiest if you want to offer some flexibility (and avoid fixed list of views via controllers/python code).
In very specific scenario (mostly snippets, designer or studio stuff), the view content needs to be read and not rendered. As the designers still have the read access on `ir.ui.view`, we kept a public method that allowed to read any view based on the external id.
A low priviledge user shouldn't read directly the view content, this is what we try to go away from.

The idea was to cover the most frequent scenario with these two methods but if specific views need to have specific access/manipulation, it should be done with a method specific for this view/scenario/module, **not** via a generic method that would open the door to read any view.

In your case, if it's just the color picker that needs to be read by every user, it's best to use another method than the generic `read_template` that is intended for users with higher priviledge (as explained in the docstring). If we start making exceptions for one view, it will be quickly a nightmare with override everywhere.
Is it possible to have a specific method/controller for the color picker without too much change?

> Side note: not convinced about the word 'asset' in that method name which is really confusing with our assets system [...] which all act differently regarding authorization and access rights, this calls for a refactoring in master ^^

The `render_*` of view is kind of a nightmare I didn't want to change more than necessary, no problem for a bit of cleaning in master. The "asset" was still the best I found but I am open to suggestion.

**Comment by qsm-odoo** [comment]
@mart-e Thanks for the detailed explanation, I understand all of this is a good idea :+1: I just wondered why the inconsistency between render and read. However, in fact, the only code which seems to still call that `read_template` method is that colorpicker. I'll create a controller to read that for stable and replace by a render in master. At the same time, should I consider removing the `read_template` method entirely in master or do we need to keep it for some compatibility stuff?

**Comment by mart-e** [comment]
> At the same time, should I consider removing the `read_template` method entirely in master or do we need to keep it for some compatibility stuff?

Mo, if we can remove that, go headed. One less confusing method ;-)

**Comment by qsm-odoo** [comment]
I guess no need for a controller as suggested in fact. Let's merge this specific patch for `read_template` for 14.0 stable version and then I'll remove the method entirely in master :+1: (I'll let forward-port bot create the PR then adapt it :+1:)

@robodoo r+
